### PR TITLE
completed routs

### PR DESCRIPTION
**Pull request by VasYerim**
completed routes and add middlewares

### CODE REVIEW

File: routes/api/contacts.js [hunk 1]
```diff
@@ -1,25 +1,78 @@
...
+    return res.status(500).json({ error: error.message });
+  }
 });
 
 router.get("/:contactId", async (req, res, next) => {
-  res.json({ message: "template message" });
+  try {
+    const id = req.params.contactId;
+    const contact = await getById(id);
+    if (!contact) {
```
Comment (VasylArtymovych) [inline routes/api/contacts.js:30]: add description

File: routes/api/contacts.js [hunk 2]
```diff
@@ -1,25 +1,78 @@
...
-  res.json({ message: "template message" });
-});
+router.put(
+  "/:contactId",
+  validateUpdateContactFields,
+  async (req, res, next) => {
+    const id = req.params.contactId;
+
+    const updatedContact = await updateContact(id, req.body);
+    if (!updatedContact) {
```
Comment (VasylArtymovych) [inline routes/api/contacts.js:71]: good job

**Review by VasylArtymovych — CHANGES_REQUESTED** [review]

**Comment by VasylArtymovych** [comment]
finish my fixing

**Comment by VasYerim** [comment]
fix my code

**Review by VasylArtymovych — APPROVED** [review]

**Pull request merged**